source/Halibut/Transport/SecureListeningClient.cs

```diff
@@ -183,23 +183,35 @@ public async Task ExecuteTransactionAsync(ExchangeActionAsync protocolHandler, C
         void HandleError(Exception? lastError, bool retryAllowed, bool hasConnected)
         {
             if (lastError == null)
+            {
                 return;
+            }
 
             lastError = lastError.UnpackFromContainers();
 
             var error = new StringBuilder();
             error.Append("An error occurred when sending a request to '").Append(ServiceEndpoint.BaseUri).Append("', ");
             error.Append(retryAllowed ? "before the request could begin: " : "after the request began: ");
             error.Append(lastError.Message);
-
-            var inner = lastError as SocketException;
-            if (inner != null)
+            
+            
+            switch (lastError)
```

Comment:
Should we do the same for the websocket client? https://github.com/OctopusDeploy/Halibut/blob/cdf0c4606b4903121fca866e203a34978a25a952/source/Halibut/Transport/SecureWebSocketClient.cs#L150